packages/api/src/controllers/organicHistoryController.js

```diff
@@ -13,118 +13,20 @@
  *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
  */
 
-const baseController = require('./baseController');
-const { transaction, Model } = require('objection');
-const model = require('../models/organicHistoryModel');
-const locationModel = require('../models/locationModel');
+
+const organicHistoryModel = require('../models/organicHistoryModel');
 
 module.exports = {
-  async addEntry(req, res) {
-    if (!locationModel.isCropEnabled(req.body.location_id)) {
-      return res.status(400).send('Location must be crop enabled.');
-    }
 
-    const trx = await transaction.start(Model.knex());
-    try {
-      const result = await baseController.postWithResponse(model, req.body, req, { trx });
-      await trx.commit();
-      res.status(201).send(result);
-    } catch (error) {
-      await trx.rollback();
-      res.status(400).json({
-        error,
-      });
-    }
+  addOrganicHistory() {
+    return async (req, res, next) => {
```

Comment:
Why does LF frequently complicate things by returning a function when it's not necessary? Is there some advantage to this?